src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureKeyVaultReference/AzureKeyVaultSecretProvider.cs

```diff
@@ -31,15 +38,19 @@ public AzureKeyVaultSecretProvider(TokenCredential credential = null, IEnumerabl
                     _secretClients[keyVaultId] = client;
                 }
             }
-        }
 
-        public async Task<string> GetSecretValue(Uri secretUri, CancellationToken cancellationToken)
-        {
-            if (secretUri == null)
+            if (certificateClients != null)
             {
-                throw new ArgumentNullException(nameof(secretUri));
```

Comment:
**Note:** I removed the `secretUri` null check here because this exception will never be thrown since `secretUri` has already been validated before calling `GetSecretValue`.
Additionally, this class should only throw exceptions which can be caught and converted to `KeyVaultReferenceException` by `AzureKeyVaultKeyValueAdapter`.
`ArgumentNullException` will bubble up to the application even if `optional` is set to true in `AddAzureAppConfiguration` method.